SCClassLibrary/Common/Control/Node.sc

```diff
@@ -141,23 +141,21 @@ Node {
 		server.sendMsg(10, nodeID);//"/n_trace"
 	}
 
-	query {
-		OSCFunc({ arg msg;
-			var cmd, argnodeID, parent, prev, next, isGroup, head, tail;
-			# cmd, argnodeID, parent, prev, next, isGroup, head, tail = msg;
-			// assuming its me ... if(nodeID == argnodeID)
-			Post
-				<< if(isGroup == 1, "Group:\t" , "Synth:\t") << nodeID << Char.nl
-				<< "parent:\t" << parent << Char.nl
-				<< "prev:\t" << prev << Char.nl
-				<< "next:\t" << next << Char.nl;
-			if(isGroup == 1) {
-				Post
-					<< "head:\t" << head << Char.nl
-					<< "tail:\t" << tail << Char.nl << Char.nl;
-			};
-		}, '/n_info', server.addr).oneShot;
-		server.sendMsg(46, nodeID) //"/n_query"
+	query { |action|
+		action = action ?? {
+			{ |cmd, argnodeID, parent, prev, next, isGroup, head, tail|
+				var group = isGroup == 1;
+				(
+					if(group, "Group: ", "Synth: ") ++ "%\nParent: %\nPrev: %\nNext: %\n" ++ if(group, "Head: %\nTail: %\n\n", "\n")
```

Comment:
this is getting a bit dense to read... i'd recommend splitting the separately concatenated statements onto separate lines:

```
if(group, "Group: ", "Synth: ")
    ++ "%\nParent: %\nPrev: %\nNext: %\n"
    ++ if(group, "Head: %\nTail: %\n\n", "\n")
```

it might be clearer as a postf, which would only require one method call and one fewer set of parens

Reply:
done!